### ISSUE TITLE
tosi: Tosi_Case4.prm SIGFPE error

### ISSUE THREAD

**Issue by tjhei** [geodynamics/aspect#3076]
when testing the benchmarks, the Tosi_Case4.prm causes a SIGFPE:
```

#0  0x000000000197c6a0 in dealii::trace<2, double> (d=...)
    at /hdscratch/heister/candi-v9.1.0-rc1/deal.II-v9.1.0-rc1/include/deal.II/base/symmetric_tensor.h:2746
#1  0x0000000001b01d5f in dealii::deviator<2, double> (t=...)
    at /hdscratch/heister/candi-v9.1.0-rc1/deal.II-v9.1.0-rc1/include/deal.II/base/symmetric_tensor.h:3270
#2  0x00007ffff7e98829 in aspect::TosiBenchmark::TosiMaterial<2>::evaluate (this=0x33d4fa0, in=..., out=...)
    at /scratch/aspect-git2/benchmarks/tosi_et_al_2015_gcubed/tosi.cc:82
#3  0x00000000016d9fbf in aspect::AdiabaticConditions::ComputeProfile<2>::initialize (this=0x31e3790)
    at ../source/adiabatic_conditions/compute_profile.cc:158
#4  0x00000000021db78f in aspect::Simulator<2>::Simulator (this=0x7fffffffa760, mpi_communicator_=1140850688, prm=...)
    at ../source/simulator/core.cc:334
#5  0x0000000001a598c6 in run_simulator<2> (
    raw_input_as_string="include Tosi_base.prm\n\nset Additional shared libraries", ' ' <repeats 12 times>, "= ./libtosi_benchmark.so\n\nset Output directory", ' ' <repeats 23 times>, "= Case4_6\n\n# Between the cases the \"Pressure viscosity parameter\""..., 
    input_as_string="include Tosi_base.prm\n\nset Additional shared libraries", ' ' <repeats 12 times>, "= ./libtosi_benchmark.so\n\nset Output directory", ' ' <repeats 23 times>, "= Case4_6\n\n# Between the cases the \"Pressure viscosity parameter\""..., output_xml=false, 
    output_plugin_graph=false, validate_only=false) at ../source/main.cc:600
#6  0x0000000001a34a22 in main (argc=2, argv=0x7fffffffd938) at ../source/main.cc:805
```
it looks like there is no strain rate (during adiabatic conditions):
```
(gdb) frame 2
(gdb) p in.strain_rate
$2 = std::vector of length 0, capacity 1
```

**Comment by tjhei** [comment]
@anne-glerum can you take a look please?

**Comment by anne-glerum** [comment]
It's on my list!

**Comment by tjhei** [comment]
fixed by #3180
